tests/generation/test_stopping_criteria.py

```diff
@@ -98,6 +99,21 @@ def test_max_time_criteria(self):
         criteria = MaxTimeCriteria(max_time=0.1, initial_timestamp=time.time() - 0.2)
         self.assertTrue(all(criteria(input_ids, scores)))
 
+    def test_eos_token_criteria(self):
+        criteria = EOSTokenCriteria(eos_token_id=0)
+
+        input_ids, scores = self._get_tensors(5)
+        input_ids[:, -1] = 0
+        self.assertTrue(all(criteria(input_ids, scores)))
+
+        input_ids, scores = self._get_tensors(5)
+        input_ids[:2, -1] = 0
+        self.assertListEqual(criteria(input_ids, scores).tolist(), [True, True, False])
```

Comment:
```suggestion
        input_ids, scores = self._get_tensors(5)
        input_ids[:2, -1] = 0
        input_ids[2, -1] = 1
        self.assertListEqual(criteria(input_ids, scores).tolist(), [True, True, False])
```
`_get_tensors` is random, I think we may get failures here if we don't hardcode the test case.

Reply:
No, because we hardcoded the last token id and the criteria looks only at the last one. Correct me if I misunderstood you

Reply:
> the criteria looks only at the last one

Agreed

> we hardcoded the last token id

In this case we hardcode the first two rows (`input_ids[:2, -1] = 0`). The third row has no hardcoded value -- while the corresponding criteria result should be `False` in most cases, nothing (AFAIK) prevents the last token in the last row to be `0`, causing this part of the test to fail.

Reply:
got it, accepted